File: backend/models/history.model.js
const mongoose = require("mongoose");

const HistorySchema = new mongoose.Schema(
  {
    childId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Child",
      required: [true, "Child ID is required"],
    },
    diagnosis: {
      type: String,
      required: [true, "Diagnosis is required"],
      trim: true,
      minlength: [3, "Diagnosis must be at least 3 characters long"],
    },
    disease: {
      type: String,
      required: [true, "Disease is required"],
      trim: true,
      minlength: [3, "Disease must be at least 3 characters long"],
    },
    treatment: {
      type: String,
      required: [true, "Treatment is required"],
      trim: true,
      minlength: [5, "Treatment must be at least 5 characters long"],
    },
    notes: {
      type: String,
      trim: true,
      maxlength: [500, "Notes cannot exceed 500 characters"],
    },
    date: {
      type: Date,
      required: [true, "Date is required"],
      validate: {
        validator: function (value) {
          return value <= new Date();
        },
        message: "Date cannot be in the future",
      },
    },
    time: {
      type: String,
      required: [true, "Time is required"],
    },
    doctorName: {
      type: String,
      required: [true, "Doctor name is required"],
      trim: true,
      default: "Dr. Islam Hasib",
    },
    notesImage: {
      type: String,
      default: null,
      validate: {
        validator: function (value) {
          return !value || /\.(jpg|jpeg|png|gif)$/i.test(value);
        },
        message: "Notes image must be a valid image file",
      },
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("History", HistorySchema);

File: lib/models/history_model.dart
import 'package:intl/intl.dart';

class History {
  final String id;
  final String diagnosis;
  final String disease;
  final String treatment;
  final String notes;
  final String notesImage;
  final DateTime date;
  final String time;
  final String doctorName;

  History({
    required this.id,
    required this.diagnosis,
    required this.disease,
    required this.treatment,
    required this.notes,
    required this.notesImage,
    required this.date,
    required this.time,
    required this.doctorName,
  });

  factory History.fromJson(Map<String, dynamic> json) {
    return History(
      id: json['_id'] ?? '',
      diagnosis: json['diagnosis'] ?? '',
      disease: json['disease'] ?? '',
      treatment: json['treatment'] ?? '',
      notes: json['notes'] ?? '',
      notesImage: json['notesImage'] ?? '',
      date: DateTime.parse(json['date'] ?? DateTime.now().toIso8601String()),
      time: json['time'] ?? '',
      doctorName: json['doctorName'] ?? 'Dr. Islam Hasib',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'diagnosis': diagnosis,
      'disease': disease,
      'treatment': treatment,
      'notes': notes,
      'notesImage': notesImage,
      'date': date.toIso8601String(),
      'time': time,
      'doctorName': doctorName,
    };
  }

  History copyWith({
    String? id,
    String? diagnosis,
    String? disease,
    String? treatment,
    String? notes,
    String? notesImage,
    DateTime? date,
    String? time,
    String? doctorName,
  }) {
    return History(
      id: id ?? this.id,
      diagnosis: diagnosis ?? this.diagnosis,
      disease: disease ?? this.disease,
      treatment: treatment ?? this.treatment,
      notes: notes ?? this.notes,
      notesImage: notesImage ?? this.notesImage,
      date: date ?? this.date,
      time: time ?? this.time,
      doctorName: doctorName ?? this.doctorName,
    );
  }
}